code/utility/palindrome/palindrome_check/palindrom_check.cpp

```diff
@@ -0,0 +1,93 @@
+#include <iterator>
+#include <type_traits>
+#include <functional>
+
+namespace palindrom_check{
+template<typename _InputIter,
+         typename _ValueNotEqualTo = std::not_equal_to<typename _InputIter::value_type>,
+         typename _IterLess = std::less<_InputIter>>
+bool
+isPalindromeRecursive(_InputIter begin, _InputIter end)
+{
+    --end;
```

Comment:
I don't know if you can check this, but a check for the distance between the iterators being 1/0, would simplify your code since you can immediately return true, in all of your implementations.

Reply:
sorry, I don't know what your mean..

Reply:
I mean, check if the container passed as an input has a length of 1 or 0, and return true in those cases. Sorry for being unclear!

Reply:
I added test for empty container, and length of 1 is the same as that length is odd

Reply:
There is no problem if you decrement an iterator in which begin equals end, right? (Just confirming, since the first thing you do is decrement the end iterator.)

Reply:
sorry, this is problem, I had fix it, but not add to commit haha, #3007